launcher/container_runner.go

```diff
@@ -153,7 +176,21 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
 		if err != nil {
 			return nil, fmt.Errorf("failed to get principal tokens: %w", err)
 		}
-		return [][]byte{[]byte(idToken)}, nil
+
+		tokens := [][]byte{[]byte(idToken)}
+
+		if len(launchSpec.ImpersonateServiceAccounts) != 0 {
+			// Fetch impersonated ID tokens.
+			for _, sa := range launchSpec.ImpersonateServiceAccounts {
```

Comment:
We don't need this if and for, just the for statement is fine, and we will just do nothing if the impersonateServiceaccounts slice is empty.